lib/statsd.js

```diff
@@ -60,12 +60,9 @@ var Client = function (host, port, prefix, suffix, globalize, cacheDns, mock, gl
     this.socket.on('error', this.errorHandler);
   }
 
+  this.cacheDns = options.cacheDns;
   if(options.cacheDns === true){
-    dns.lookup(options.host, function(err, address, family){
-      if(err == null){
-        self.host = address;
-      }
-    });
+    this.resolveAddress();
```

Comment:
this is javascript crap.  you can't refer to `this` in the callback.

No, it won't block send, which will still use the unresolved host.